services/web/server/src/simcore_service_webserver/catalog/client.py

```diff
@@ -39,7 +39,14 @@ def _handle_client_exceptions(app: web.Application) -> Iterator[ClientSession]:
 
         yield session
 
-    except (asyncio.TimeoutError, ClientConnectionError, ClientResponseError) as err:
+    except ClientResponseError as err:
+        if err.status == 404:
```

Comment:
MINOR: ``` if err.status == web.HTTPNotFound.status_code``

THOUGHT: 404s have some issues, specially on APIs like the web-api that has several routing services in front. Here is [a good description](https://github.com/stickfigure/blog/wiki/How-to-%28and-how-not-to%29-design-REST-APIs#rule-8-dont-use-404-to-indicate-not-found).  It is a good moment to read. we can follow up on that at another point in time..

Reply:
In case you go full on this thing @pcrespov @matusdrobuliak66 I would like to discuss it. in our case we control the complete pipeline, so I would not like that we now start putting whatever fancy and changing responses all over the place... there is nothing more confusing than having some old parts using 404 and new parts that use something else. so please no mix. if we change we change all, otherwise I'm against it.

Reply:
:+1: We can discuss, for now I will keep it simple